app/src/main/java/com/makuno/memory/ui/game/viewmodel/GameViewModel.kt

```diff
@@ -110,9 +103,37 @@ internal class GameViewModel
         resetTimer()
     }
 
-    fun saveScore(score: Score) {
+    private fun saveScore(score: Score) {
         viewModelScope.launch {
             productRepository.saveScore(score)
         }
     }
+
+    fun compareCards(
+        gameCard: GameCard,
+        easyFlipView: EasyFlipView,
+        gameCardMatchListener: GameCardMatchListener
+    ) {
+        when {
+            firstGameCard == null && firstEasyFlipView == null -> {
+                firstGameCard = gameCard
+                firstEasyFlipView = easyFlipView
+            }
+            firstGameCard != null && gameCard == firstGameCard -> {
+                firstGameCard = null
+                firstEasyFlipView = null
+
+                addOneToScore()
+                gameCardMatchListener.onCardsSimilar()
+            }
+            firstGameCard != null && gameCard != firstGameCard -> {
+                val initialFlipView = firstEasyFlipView
+
+                firstGameCard = null
+                firstEasyFlipView = null
+
+                gameCardMatchListener.onCardsDifferent(initialFlipView, easyFlipView)
+            }
```

Comment:
You have one particular recurring boolean which is `firstGameCard != null`, only the first statement in the when is different so what you should is use an if statement to check if 
`firstGameCard == null &&  && firstEasyFlipView == null` then  put this statement there and the when statement excluding the first option should be in the else
` {
                firstGameCard = gameCard
                firstEasyFlipView = easyFlipView
            }`

Reply:
Noted. Will update the statements.